libminifi/include/utils/expected.h

```diff
@@ -0,0 +1,176 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <type_traits>
+#include <utility>
+#include "nonstd/expected.hpp"
+#include "utils/detail/MonadicOperationWrappers.h"
+#include "utils/GeneralUtils.h"
+#include "utils/meta/detected.h"
+
+namespace org::apache::nifi::minifi::utils {
+namespace detail {
+
+template<typename T>
+inline constexpr bool is_expected_v = false;
+
+template<typename V, typename E>
+inline constexpr bool is_expected_v<nonstd::expected<V, E>> = true;
+
+// map implementation
+template<typename Expected, typename F, typename = std::enable_if_t<is_expected_v<utils::remove_cvref_t<Expected>>>>
+auto operator|(Expected&& object, map_wrapper<F> f) {
```

Comment:
I didn't bother with noexcept as it would've required even more metaprogramming, I would've needed to basically duplicate the implementation inside the noexcept specifier. There is a lot of code duplication inside the implementation, but I think this is the simplest way to do it. I could use `if constexpr` instead of SFINAE a lot, so I think it should also compile faster, but I don't want to implement them twice to measure.